pkg/build/registry/buildlog/rest.go

```diff
@@ -63,29 +63,41 @@ func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.
 	if !ok {
 		return nil, errors.NewBadRequest("did not get an expected options.")
 	}
+	// TODO: Validate BuildLogOptions
 	obj, err := r.Getter.Get(ctx, name)
 	if err != nil {
 		return nil, err
 	}
 	build := obj.(*api.Build)
+	if buildLogOpts.Previous {
+		version := butil.VersionForBuild(build)
+		// Use the previous version
+		version--
```

Comment:
As a matter of fact I had that check but in the client you either have run `oc logs -p bc/cfg` or `oc logs -p build/cfg-n` whereas in the server we don't know what was it (we always resolve to a build in the client before proceeding with the request). So the following GET will return the same best message I could have here.